applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -1122,6 +1317,67 @@ namespace Kratos
     //************************************************************************************
     //************************************************************************************
 
+    void BaseSolidElement::CalculateAndAddKm(
```

Comment:
The function was previously defined but empty, and besides, it is always always the same code repeated I think that @pooyan-dadvand agrees with me that we should not duplicate code